### DISCUSSION TITLE
[TS] Generating through OpenAI SDK directly

### DISCUSSION THREAD

**Discussion by louis030195** [langfuse/langfuse#1306] [Ideas]
### Describe the feature or potential improvement

```ts
import { Langfuse } from "langfuse";
 
const langfuse = new Langfuse();
 
// Example generation creation
const generation = trace.generation({
  name: "chat-completion",
  model: "gpt-3.5-turbo",
  modelParameters: {
    temperature: 0.9,
    maxTokens: 2000,
  },
  input: messages,
});
 
// Application code
const chatCompletion = await llm.respond(prompt);
 
// End generation - sets endTime
generation.end({
  output: chatCompletion,
});

```

to:

```ts

const langfuse = new Langfuse();
 const openai = new OpenAI();


const openai = await openai.completion(prompt);
```

**Comment by marcklingen** [comment]
Agree, the integration for OpenAI TS could be way better. Adding your +1 to #1065

Related, a more generalized wrapper to automatically instrument functions would be awesome for TS. Currently we are working on decorators for Python (#1009) and this could be a great next step.

**Comment by hassiebp** [comment]
@louis030195 We've just released the JS integration for the OpenAI SDK in the [Langfuse JS SDK 3.6.0](https://github.com/langfuse/langfuse-js/releases/tag/v3.6.0) 🎉 Feel free to try it out and let us know your feedback 🙏🏾

[Link to release](https://github.com/langfuse/langfuse-js/releases/tag/v3.6.0)
[Link to docs](https://langfuse.com/docs/integrations/openai/js/get-started)
